exercises/practice/list-ops/.meta/proof.ci.ts

```diff
@@ -10,17 +10,17 @@ const Null: Cons = {
     return this.value
   },
 
-  push(item): Cons {
+  push<T>(item: T): Cons<T> {
     // eslint-disable-next-line @typescript-eslint/no-use-before-define
-    return new Cons(item, this)
+    return new Cons(item, this) as Cons<T>
```

Comment:
Why is this necessary? Doesn't `new Cons(item, this)` already infer `Const<T>`? If not, shouldn't it be `new Cons<T>(item, this)`?

Reply:
In this case `Cons` would be of type `Cons<T | undefined>`, however because of how `Null` works the `undefined` part can be omitted, and since is of type `Cons<T | undefined>` using `new Cons<T>(item, this)` throws an error on TypeScript.